src/lib/crosslink.js

```diff
@@ -0,0 +1,9 @@
+import { compressToBase64 } from 'lz-string';
+
+const version = 'lz';
+
+export function generateCrosslink(options) {
+  const base = JSON.stringify(options);
+  const compressed = compressToBase64(base);
```

Comment:
I've been tested a some different formats for reduce url size (indcludes binary format), but main problem it's a lot of spaces and same words into code. And better way to compress it's general compress like lz. Some examples i've tested:
 * clear base64
 * [message pack](https://msgpack.org/index.html)
 * lz + base64
It looks like that
<img width="1484" alt="Screen Shot 2021-10-05 at 20 19 56" src="https://user-images.githubusercontent.com/242514/140922719-f3f288aa-a98f-4547-ac06-ee07ee0377a9.png">
:

Reply:
May be it's a good idea trim code before pack and reformat in play, but it's may be broke some buzzlers.